src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.ILLink.targets

```diff
@@ -20,6 +20,11 @@ This file contains the .NET 5-specific targets to customize ILLink
           Condition=" '$(AndroidLinkMode)' == 'Full' and '%(ResolvedFileToPublish.Extension)' == '.dll' and '%(ResolvedFileToPublish.AssetType)' != 'native' "
           TrimMode="link"
       />
+      <!-- TODO: remove setting the trim mode here, once the support packages are updated to NET6 and compatability packages not needed -->
+      <ResolvedFileToPublish
+          Condition=" '$(AndroidLinkMode)' == 'SdkOnly' and ( $([System.String]::Copy(%(Filename)).StartsWith ('System.')) or $([System.String]::Copy(%(Filename)).StartsWith ('Xamarin.Android')) ) ">
```

Comment:
Should we make this a little more specific and check if it starts with:
* `Xamarin.Android.Support.`
* `Xamarin.AndroidX.`
* `Xamarin.Google.` - https://www.nuget.org/packages/Xamarin.Google.Android.Material/
* `Xamarin.GooglePlayServices.`

Reply:
I'm not sure using `System` is a good idea here. Users can name assemblies what every they want, so `System.Foo.UI` is a perfectly valid assembly name. 

 is there existing Metadata we can leverage to detect a) system assemblies (like FrameworkFileName? ) and support packages. 

for example, can we get away with using the `_AndroidAssemblySkipCases`  item group from https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Xamarin.Android.SkipCases.projitems#L24 to drive this.

Reply:
I think @radekdoulik is trying to get assemblies like these that are compat packages on NuGet:

https://github.com/xamarin/net6-samples/blob/c4a4f4a38f062f385bd4ed98602e8d9f9764b33e/HelloForms/Directory.Build.props#L70-L86

I don't think you can use `%(FrameworkFileName)`, because these are normal NuGet packages.

Reply:
It is temporary solution, so I didn't want to complicate it much. OTOH, who knows how long it will stay that way, so I will refine it a bit.

Maybe I will just use a list of known assemblies?

Reply:
If you look at `Xamarin.GooglePlayServices` or `Xamarin.AndroidX`, there are a lot! It might be a couple hundred.

Reply:
OK, made it mode strict about support libraries names.

For compat assemblies I will create a list of names. Probably from typeforwarders collected from `mscorlib`.

I will put that list to the next PR.